llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -552,8 +552,23 @@ PPCTargetLowering::PPCTargetLowering(const PPCTargetMachine &TM,
     setOperationAction(ISD::UINT_TO_FP, MVT::i32, Legal);
 
     // SPE supports signaling compare of f32/f64.
-    setOperationAction(ISD::STRICT_FSETCCS, MVT::f32, Legal);
-    setOperationAction(ISD::STRICT_FSETCCS, MVT::f64, Legal);
+    // But it doesn't comply IEEE-754 rules for comparing
+    // special values like NaNs, Infs.
+    if (!getTargetMachine().Options.NoNaNsFPMath ||
+        !getTargetMachine().Options.NoInfsFPMath) {
```

Comment:
Don't introduce new uses of the global options. They are in the process of removal. You could instead make these custom, check the fast math flags, and then defer to the default expansion to libcall